scripts/xtensa-build-zephyr.py

```diff
@@ -186,6 +187,16 @@ def parse_args():
 sign must be used (https://bugs.python.org/issue9334)""",
 	)
 
+	parser.add_argument("--key_name", required=False,
+			    default = "community",
```

Comment:
`default="community"`, don't insert SPACE before and after `=` for key word arguments. same for the `default=False` in 
--output_dir_per_platform option.

Any reason that you don't want a short option for --key_name and --output_dir_per_platform?

you can make --key_name a choice parameter, with `community` as it's default value. example:
`parser.add_argument('move', choices=['rock', 'paper', 'scissors'])`, so that only the listed values are allowed for the option. and you don't need the `validate_key_name()` anymore.

Reply:
@marc-hb @aiChaoSONG Unfortunately we cannot use `--key-type` or `--key-subdir` for a string parameter.
```
print(args.key-type)
TypeError: unsupported operand type(s) for -: 'NoneType' and 'type'

print(args.key-subdir)
NameError: name 'subdir' is not defined
```

I think we can use `--key_type`, it works. In xtensa-build-zephyr.py, variables are using `_` in their names. I think `-` can be taken as operator and so it's not safe.

Reply:
Agree that we can provide a candidate value list for --key_type.

 ```--key_type``` provides some file directory naming convention way. The user would better ensure he/she is using the correct key with corresponding --key_type specified when using option ```-k```  to use the debugkey or any other key he/she wants to use correctly.

Example:
```
./scripts/xtensa-build-zephyr.py -i IPC4 -k ~/xcc/meu_tgl/private.pem --key_type "dbgkey"  tgl
```

 (I haven't tried this PR with the IPC4 option yet , but it should work

Reply:
> I also want to add an option `--outdir-root-name` (default `sof`) so that we can use `sof-ipc4` as the root dir name for output. Is it good? @marc-hb @aiChaoSONG @XiaoyunWu6666

I'm OK with that , though the increasing number of option may make the usage of this script more complicated.
But as long as accurate help information is provided , it's good cause more freedom is provided.

Reply:
> Unfortunately we cannot use --key-type or --key-subdir for a string parameter.

@mengdonglin I think this may be a bug or requirement of argparse. When people use parser.add_argument, they tend to use hyphen(-), but for the parsed arg, you need to use underscore to replace the hyphen.

example: 	if an argument is added with parser.add_argument("--key-name", ...), you need to use args.key_name to refer to the option added. you can print args to verify this.

Reply:
> Unfortunately we cannot use --key-type or --key-subdir for a string parameter.

Yes you can, as @aiChaoSONG explained. See `--zephyr-ref` and `zephyr_ref` in this script and other options in pretty much every Python script using `argparse`. Please stick to common practice.

Reply:
> > > I also want to add an option --outdir-root-name (default sof) so that we can use sof-ipc4 as the root dir name for output. Is it good?
> 
> > I'm OK with that , though the increasing number of option may make the usage of this script more complicated.
> 
> I don't think a rootdir option is very useful because the staging directory has to copied somewhere else anyway and I agree that it increases the script's complexity.

@marc-hb @XiaoyunWu6666 Thanks for sharing your idea! Then I will not introduce a new option for this.